Rx/v2/src/rxcpp/operators/rx-reduce.hpp

```diff
@@ -128,23 +138,39 @@ struct reduce : public operator_base<rxu::value_type_t<reduce_traits<T, SourceOp
             reduce_state_type& operator=(reduce_state_type o) RXCPP_DELETE;
         };
         auto state = std::make_shared<reduce_state_type>(initial, std::move(o));
+        auto on_completed = [state]() {
+            rxu::maybe<value_type> result;
+            try {
```

Comment:
rxcpp::util has an on_exception function that should be used instead of a manual try/catch whenever possible.